src/evalfunc.c

```diff
@@ -3153,6 +3172,21 @@ check_internal_func(int idx, int argcount)
     return -1;
 }
 
+/*
+ * Check arguments for functions that can't handle :type/:class.
+ */
+    static int
+check_args_for_type(int idx, int argcount, typval_T *argvars)
+{
+    if (!func_allows_type(idx))
+    {
+	for (int i = 0; i < argcount; ++i)
+	    if (check_typval_is_value(&argvars[i]) == FAIL)
+		return FCERR_OTHER;
+    }
+    return FCERR_NONE;
```

Comment:
The FCERR_xxx values are used for error codes when calling a function.  Can you change this function to return OK or FAIL?

Reply:
Changed. (I thought checking args was part of calling a function).